Solutions/CohesitySecurity/Data Connectors/Helios2Sentinel/IncidentProducer/IncidentProducer.cs

```diff
@@ -0,0 +1,252 @@
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json;
+using StackExchange.Redis;
+using System.Collections.Generic;
+using System.Collections;
+using System.IO;
+using System.Net.Http.Headers;
+using System.Net.Http;
+using System.Net;
+using System.Text.Json;
+using System.Text;
+using System.Threading.Tasks;
+using System;
+using System.Dynamic;
+using Newtonsoft.Json.Converters;
+using Microsoft.WindowsAzure.Storage;
+using Helios2Sentinel;
+
+namespace Helios2Sentinel
+{
+    public class IncidentProducer
+    {
+        private static readonly object queueLock = new object();
+        private static Lazy<ConnectionMultiplexer> lazyConnection = CreateConnection();
+        private static string containerName = Environment.GetEnvironmentVariable("containerName");
+        private static string blobStorageConnectionString = Environment.GetEnvironmentVariable("BlobStorageConnectionString");
+        public static long GetPreviousUnixTime(ILogger log)
+        {
+            DateTime previousDateTime = DateTime.Now;
+            try
+            {
+                previousDateTime = previousDateTime.AddDays(long.Parse(Environment.GetEnvironmentVariable("startDaysAgo")));
+            }
+            catch  (Exception ex)
+            {
+                previousDateTime = previousDateTime.AddDays(-30);
+                log.LogError("Exception --> 1 " + ex.Message);
+            }
+            return ((DateTimeOffset)previousDateTime).ToUnixTimeMilliseconds() * 1000;
+        }
+
+        public static long GetCurrentUnixTime()
+        {
+            return ((DateTimeOffset)DateTime.Now).ToUnixTimeMilliseconds() * 1000;
+        }
+
+        public static ConnectionMultiplexer Connection
+        {
+            get
+            {
+                return lazyConnection.Value;
+            }
+        }
+
+        private static Lazy<ConnectionMultiplexer> CreateConnection()
+        {
+            return new Lazy<ConnectionMultiplexer>(() =>
+            {
+                return ConnectionMultiplexer.Connect(Environment.GetEnvironmentVariable("connectStr"));
+            });
+        }
+
+        public static async Task ParseAlertToQueue(
+            [Queue("%CohesityQueueName%"), StorageAccount("AzureWebJobsStorage")] ICollector<string> outputQueueItem,
```

Comment:
Why did we decide to put the queue name to the environment variable? If the deployment wizard creates a storage account with a unique name, then we can just have a common constant between producer and consumer for the queue name